nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -105,7 +105,7 @@ let
   upstreamConfig = toString (flip mapAttrsToList cfg.upstreams (name: upstream: ''
     upstream ${name} {
       ${toString (flip mapAttrsToList upstream.servers (name: server: ''
-        server ${name} ${optionalString server.backup "backup"};
+        server ${name} ${optionalString (server.failTimeout != "10") "fail_timeout=${server.failTimeout}"} ${optionalString server.backup "backup"};
```

Comment:
Shouldn't we introduce a generic option to add potential other arguments?

Reply:
Would there be any confusion with `${upstream.extraConfig}`

Reply:
Addede `extraOptions`.